libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.resolver.ts

```diff
@@ -44,4 +45,17 @@ export class NationalRegistryXRoadResolver {
       user.authorization,
     )
   }
+
+  @Query(() => NationalRegistryResidenceHistory, {
```

Comment:
Would like to see this field as a field resolver. This means that we could query this field on the nationalRegistryPerson field:
```graphql
query {
  nationalRegistryPerson {
    id
    residence {
	  address {
	    postalCode
	  }
	  country
	  history {
	    address {
	      postalCode
	    }
	    country
	  }
    }
  }
}
```

where `history` of the residence field would be resolved in a field resolver. Leads to a better structured object.

Reply:
great idea